libraries/ms-drivers/test/test_pca9539r_gpio_expander.c

```diff
@@ -308,3 +310,17 @@ void test_pca9539r_gpio_get_state_invalid_address(void) {
   Pca9539rGpioState state;
   TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, pca9539r_gpio_get_state(&address, &state));
 }
+void prv_test(const struct GpioAddress *address, void *context) {
+  GpioState state = GPIO_STATE_LOW;
+  gpio_get_state(address, &state);
+}
+void test_pca9539r_gpio_subscribe_interrupts(void) {
```

Comment:
This test is a good start, but you should do more than just test that `pca9539r_gpio_subscribe_interrupts` returns OK; you should test that it actually does everything it's supposed to. I think a good way to do that would be to check that the callback you register is called when you trigger an interrupt on the pin with `gpio_it_trigger_interrupt`.

A way to do that would be to have a static variable (maybe a `uint8_t s_times_interrupt_callback_called`), reset it to 0 in `setup_test`, and increment it in your callback. Then in this test you can call `gpio_it_trigger_interrupt` and test that the callback was called by testing that your static variable was incremented.